internal/k8s/crdcontroller.go

```diff
@@ -250,11 +256,14 @@ func (c *AviController) SetupAKOCRDEventHandlers(numWorkers uint32) {
 				aviinfra := obj.(*akov1alpha1.AviInfraSetting)
 				namespace, _, _ := cache.SplitMetaNamespaceKey(utils.ObjKey(aviinfra))
 				key := lib.AviInfraSetting + "/" + utils.ObjKey(aviinfra)
+				statusBefore := aviinfra.Status.Status
 				if err := validateAviInfraSetting(key, aviinfra); err != nil {
 					utils.AviLog.Warnf("Error retrieved during validation of AviInfraSetting: %v", err)
 				}
+				// It's a reference, so pointer should be able to access the updated status.
+				statusAfter := aviinfra.Status.Status
 				ok, resVer := objects.SharedResourceVerInstanceLister().Get(key)
-				if ok && resVer.(string) == aviinfra.ResourceVersion {
+				if ok && resVer.(string) == aviinfra.ResourceVersion && statusAfter == statusBefore {
```

Comment:
Should we do similar changes to Istio CRDs?

Reply:
Istio CRDs may not have this problem. We can tackle it in `master` during Istio implementation. Istio's code base in layer 2 won't look for an `Accepted` or `Rejected` decision to apply the CRD, hence I don't think the problem should arise.